react-components/src/actions/cart/actions.js

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import parseError from '../../utils/parseError';
+
+/**
+ * Adds an item to the cart. If the cart doesn't exist then it's created
+ *
+ * @param {Object} payload a parameters object with the following structure:
+ *      cartDetailsQuery - the query object to execute
+ *      createCartMutation - the mutation to execute to create the cart
+ *      addToCartMutation - the mutation to execute to add the item to the cart
+ *      dispatch - the dispatch callback for the cart context
+ *      cartItems - the items to add to the cart
+ *      cartId - the id of the cart in which to add the items. This could be `undefined` or `null`
+ */
+
+export const addItemToCart = async payload => {
```

Comment:
In the `addItemToCart` action you take care of the loading dispatches, while for `removeItemFromCart`, `removeCoupon` and `addCoupon`, this is done in the `cartInitializer`. Can we somehow harmonise this?

Reply:
You're right, we should avoid dispatching UI state changes from the actions themselves. I moved the dispatch call to `useMinicart`